<?xml version="1.0"?>
<launch>
  <param name="/score_debug" value="false"/>

  <param name="/hfov" value="1.5708"/>
  <param name="/c_x" value="320.5"/>
  <param name="/c_y" value="240.5"/>
  <param name="/f_x" value="319.99"/>
  <param name="/f_y" value="319.99"/>

  <param name="/pcl_max_range" value="8.0"/>
  <param name="/depth_topic" value="/d455/depth/image_raw"/>
  <param name="/pcl_topic" value="/os_cloud_node/points"/>
  
  <param name="/depth_base" value="d455/camera_link"/>
  <param name="/pcl_base" value="os0_lidar"/>
  <param name="/body_base" value="base_body"/>
  <param name="/fixed_frame" value="map"/>

  <param name="/octomap_topic" value="/octomap"/>
  <param name="/octomap_resolution" value="0.3"/>
  <param name="/octomap_hit" value="0.9"/>
  <param name="/octomap_miss" value="0.4"/>
  <param name="/octomap_hz" value="5.0"/>
  <param name="/search_depth" value="15"/> <!-- max: 16, 0=default(max), min:1 -->

  <param name="/unknown_score_local" value="0.09"/>
  <param name="/free_score_local" value="0.11"/> <!-- lambda -->

  <param name="/collision_r" value="0.4"/>


  <!-- Nodes  -->

  <!-- main code: traversability check, obstacle check, trajectory generate, select  -->
  <node name="trot_qr" pkg="trot-q" type="trot-q_node" output="screen"/>
                      <!-- launch-prefix="gdb -ex run --> <!--args" -->
  <!-- High-level control code (MPC) -->
  <!-- <node name="mpc-control" pkg="trot-q" type="mpc-slsqp.py"/> -->

  <!-- tf publisher  -->
  <node name="tf_broadcaster" pkg="trot-q" type="tf_broadcaster.py"/>
  <!-- SLAM: FAST-LIO2 -->
  <include file="$(find trot-q)/launch/include/mapping_ouster64.launch"/>

</launch>